gpfit/plot_fit.py

```diff
@@ -0,0 +1,45 @@
+from gpfit.fit import fit
+from gpfit.print_fit import print_MA, print_SMA
+import numpy as np
+import matplotlib.pyplot as plt
+
+def plot_fit_1d(x, y, K=1, fitclass='MA'):
+    "Finds and plots a fit (MA or SMA) for 1D data"
+
+    u = np.log(x)
+    w = np.log(y)
+
+    params, _ = fit(u, w, K, fitclass)
+
+    xx = np.linspace(min(x), max(x), 1000)
+    if fitclass == 'MA':
+        A = params[[i for i in range(K*2) if i % 2 != 0]]
+        B = params[[i for i in range(K*2) if i % 2 == 0]]
+        YY = []
+        for k in range(K):
+            YY += [np.exp(B[k])*xx**A[k]]
+        stringlist = print_MA(A, B, 1, K)
+
+
+    if fitclass == 'SMA':
+        alpha = 1./params[-1]
+        A = params[[i for i in range(K*2) if i % 2 != 0]]
+        B = params[[i for i in range(K*2) if i % 2 == 0]]
+
+        yy = 0
+        for k in range(K):
+            yy += (np.exp(alpha*B[k])*xx**(alpha*A[k]))
+        YY = [yy**(1./alpha)]
```

Comment:
instead of recalculating these, it would be much cleaner to directly call `max_affine`, `softmax_affine`, and `implicit_softmax_affine`!